libraries/AP_RCProtocol/AP_RCProtocol_MavlinkRadio.cpp

```diff
@@ -0,0 +1,33 @@
+
+#include "AP_RCProtocol_config.h"
+
+#if AP_RCPROTOCOL_MAVLINK_RADIO_ENABLED
+
+#include "AP_RCProtocol_MAVLinkRadio.h"
+
+// constructor
+AP_RCProtocol_MAVLinkRadio::AP_RCProtocol_MAVLinkRadio(AP_RCProtocol &_frontend) :
+    AP_RCProtocol_Backend(_frontend)
+{}
+
+// update function to populate input
+void AP_RCProtocol_MAVLinkRadio::update(void)
+{
+    if (frontend.mavlink_radio.rc_channels_updated) {
+        frontend.mavlink_radio.rc_channels_updated = false;
+
+        uint8_t count = frontend.mavlink_radio.rc_channels.count;
+        if (count >= MAX_RCIN_CHANNELS) count = MAX_RCIN_CHANNELS;
+
+        uint16_t rc_chan[MAX_RCIN_CHANNELS];
+        for (uint8_t i = 0; i < count; i++) {
+            rc_chan[i] = ((int32_t)frontend.mavlink_radio.rc_channels.channels[i] * 5) / 32 + 1500;
```

Comment:
A comment explaining this maths would be nice

Reply:
nice. done.